airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -1670,7 +1670,7 @@ def get_dataset(self, dataset_id, project_id=None):
 
         return dataset_resource
 
-    def get_datasets_list(self, project_id=None):
+    def get_datasets_list(self, project_id=None, max_results=None, all=False):
```

Comment:
What do you think about adding the `@GoogleCloudBaseHook.fallback_to_default_project_id``decorator?` This will allow you to delete the repeating code, but also add validation - this field is required.

Reply:
I can but it will be less esasier to switch project than with parameter.